### PR TITLE
Feature/evm module

### LINKED ISSUE #473: evmOS integration

### PR DESCRIPTION
**Pull request by mn13**
## Summary by CodeRabbit

- **New Features**
	- Integration of Ethereum Virtual Machine (EVM) capabilities into the application.
	- Introduction of a new currency denomination, "award," replacing "uward."
	- Enhanced configuration options for EVM and Fee Market settings.
	- Improved transaction handling and key management features.

- **Bug Fixes**
	- Resolved dependency issues to improve compatibility within the Evmos ecosystem.

- **Documentation**
	- Updated CHANGELOG.md to reflect recent changes and enhancements.
	- Adjusted key derivation path in documentation to align with Ethereum standards.

- **Chores**
	- Dependency updates and removals for improved performance and compatibility.

### CODE REVIEW

**Comment by mn13** [comment]
Hey @MalteHerrmann, here is a draft PR to track our integration progress

File: go.mod [hunk 1]
```diff
@@ -3,6 +3,12 @@ module github.com/warden-protocol/wardenprotocol
 go 1.22.5
 
 replace (
+	// evmos integration - need this replace when importing cosmos/rosetta pkg
+	cosmossdk.io/core => cosmossdk.io/core v0.11.0
+	// evmos integration
+	github.com/cosmos/cosmos-sdk => github.com/warden-protocol/cosmos-sdk v0.50.9-evmos-warden
+	// evmos + types for slinky
+	github.com/ethereum/go-ethereum => github.com/warden-protocol/go-ethereum v1.10.26-warden-evmos-rc2
```
Comment (pitasi) [inline go.mod:11]: sad that we have to maintain these forks 😭

File: warden/app/oracle.go [hunk 2]
```diff
@@ -238,6 +240,34 @@ func createSlinkyUpgrader(app *App) AppUpgrade {
 			if err != nil {
 				return nil, fmt.Errorf("failed to set x/marketmap params: %w", err)
 			}
+
+			err = app.EvmKeeper.SetParams(
```
Comment (pitasi) [inline warden/app/oracle.go]: this is inside the v0.4 migration code (already executed in buenavista), so this code probably don't belong here

**Review by pitasi — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (mn13) [reply go.mod:11]: we discussed with @MalteHerrmann that probably we can merge our changes in their fork

File: warden/app/oracle.go (hunk 2 above)
Comment (mn13) [reply warden/app/oracle.go]: Removed upgrade part at all as we plan to start this version from scratch

**Comment by pitasi** [comment]
there are conflicts

in any case i'm waiting a bit on this, because i want to understand what needs to be done on spaceward to get it back to work with the new signature scheme

**Comment by pitasi** [comment]
> there are conflicts
> 
> in any case i'm waiting a bit on this, because i want to understand what needs to be done on spaceward to get it back to work with the new signature scheme

spaceward now works, see #764 

we can proceed with merging this

**Review by pitasi — CHANGES_REQUESTED** [review]
please search and replace these:

- `warden10kmgv5gzygnecf46x092ecfe5xcvvv9r870rq4` -> `warden1d652c9nngq5cneak2whyaqa4g9ehr8pstxj0r5`

- `m/44'/118'/0'/0/0` -> `m/44'/60'/0'/0/0`

- the chain id in `cmd/wardenkms/wardenkms.go` and `cmd/faucet-v2/main.go`

**Review by pitasi — DISMISSED** [review]
LGTM!

**Review by pitasi — APPROVED** [review]

**Pull request merged**